### PR TITLE
Read azure test response from json

### LINKED ISSUE #83: Compare cloud provider responses in unit tests
Currently we are using the pre defined response in our cloud provider unit tests. 

TODO:
- [x] read gcp response from a file 
- [x] read azure response from a file
- [x] read aws response from a file

https://github.com/redhatcloudx/rhelocator/blob/3b24b2e345bc8dc17331f82e723da109c46dd046/tests/update_images/test_azure.py#L30-L47

We want to read a raw http response from a file. That way test responses can be updated independently.

## current architecture
image

## proposed architecture
image

### PR DESCRIPTION
**Pull request by miyunari**
This PR does not include testing for the detailed azure image information. I would create an extra issue for this, if thats ok? :)

Closes #83 #80

### CODE REVIEW

File: tests/update_images/test_azure.py [hunk 1]
```diff
@@ -246,23 +248,19 @@ def test_get_all_images(mock_azure_image_versions, mock_azure_image_details):
...
 def test_format_image():
-    """Test transforming a single Azure image into a schema approved format."""
-    mocked_image = {
-        "architecture": "x86_64",
-        "hyperVGeneration": "v2",
-        "offer": "offer",
-        "publisher": "publisher",
-        "sku": "sku",
-        "urn": "publisher:offer:sku:7.6.2020082423",
-        "version": "7.6.2020082423",
```
Comment (miyunari) [inline tests/update_images/test_azure.py:257]: Any idea why the date is missing in the azure data dump we get from the cli tool? :thinking:

Comment (FKolwa) [reply tests/update_images/test_azure.py:257]: Sorry misread date for data at first (it's friday) :p 
Which CLI tool are referring to?

Comment (FKolwa) [reply tests/update_images/test_azure.py:257]: The mocked_image data above is what the format_image method expects (which itself is a product of two methods 'get_image_versions' and 'get_image_details'). But this raises a good point: I think the format_image methods should be private as it requires image data that is pre-processed. Additionally: Only AWS currently calls format_image everything else uses format_images. I think I'll create a  housekeeping task for this.

**Review by FKolwa — APPROVED** [review]
I like the changes and I'm okay with your idea to add the tests in a follow up task.

File: tests/update_images/test_azure.py (hunk 1 above)
Comment (miyunari) [reply tests/update_images/test_azure.py:257]: Ah great! Thank you for your explanation! :)

**Pull request merged**